api/swagger/server/restapi/certHandler.go

```diff
@@ -0,0 +1,222 @@
+package restapi
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"flag"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+var userAndHostname string
+
+const rootName = "rootCA.pem"
+const rootKeyName = "rootCA-key.pem"
+
+type mkcert struct {
+	ecdsa  bool
+	CAROOT string
+	caCert *x509.Certificate
+	caKey  crypto.PrivateKey
+}
+
+func (m *mkcert) Run(serverName string) ([]byte, crypto.PrivateKey, error) {
+	// Get the CA root certificate
+	m.CAROOT = getCAROOT()
+	if m.CAROOT == "" {
+		log.Fatalln("ERROR: failed to find the default CA location, set one as the CAROOT env var")
+	}
+	fatalIfErr(os.MkdirAll(m.CAROOT, 0755), "failed to create the CAROOT")
+	// Load the CA certificate and key
+	m.loadCA()
+
+	if len(serverName) == 0 {
+		flag.Usage()
+		panic("error: The String ServerName is Empty.")
+	}
+
+	// Generate keys and certificate
+	priv, err := m.generateKey(false)
+	fatalIfErr(err, "failed to generate certificate key")
+	pub := priv.(crypto.Signer).Public()
+
+	expiration := time.Now().AddDate(2, 3, 0)
+
+	tpl := &x509.Certificate{
+		SerialNumber: randomSerialNumber(),
+		Subject: pkix.Name{
+			CommonName:   serverName,
+			Organization: []string{"thyra dynamically generated"},
+		},
+
+		NotBefore:   time.Now(),
+		NotAfter:    expiration,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:    x509.KeyUsageDigitalSignature,
+	}
+
+	tpl.DNSNames = append(tpl.DNSNames, serverName)
+
+	// Create the certificate
+	cert, err := x509.CreateCertificate(rand.Reader, tpl, m.caCert, pub, m.caKey)
+
+	return cert, priv, err
+}
+
+// Get the CA root certificate depending on your OS
+func getCAROOT() string {
+	if env := os.Getenv("CAROOT"); env != "" {
+		return env
+	}
+	var dir string
+	switch {
+	case runtime.GOOS == "windows":
+		dir = os.Getenv("LocalAppData")
+	case os.Getenv("XDG_DATA_HOME") != "":
+		dir = os.Getenv("XDG_DATA_HOME")
+	case runtime.GOOS == "darwin":
+		dir = os.Getenv("HOME")
+		if dir == "" {
+			return ""
+		}
+		dir = filepath.Join(dir, "Library", "Application Support")
+	default: // Unix
```

Comment:
I don't think it's a good idead to use `default` for "`Unix`" since it might be something else than a `Unix` OS.

Reply:
If we add a `case` for `linux` (I don't think `Unix` exists since MacOS is a member of the `Unix` family, as Linux, Android, BSD... [source](https://fr.wikipedia.org/wiki/Unix)), I think the `default` should return an error as explained in [my other comment](https://github.com/massalabs/thyra/pull/359#discussion_r1052353710).

Reply:
I think this is useless because , there won't be other choice. This is using fileDirectory from specific OS and there is no others OS  which can be used with this process. There is no Default because we set all the OS available. 
For me it's useless